public/javascripts/application.js

```diff
@@ -439,19 +439,24 @@ $j(document).ready(function() {
       type: 'POST',
       url: '/kudos.js',
       data: jQuery('#new_kudo').serialize(),
-      error: function(jqXHR, textStatus, errorThrown ) {
+      error: function(jqXHR, textStatus, errorThrown) {
         var msg = 'Sorry, we were unable to save your kudos';
-        var data = $j.parseJSON(jqXHR.responseText);
 
-        if (data.errors && (data.errors.ip_address || data.errors.user_id)) {
-          msg = "You have already left kudos here. :)";
-        }
+        // When we hit the rate limit, the response is a plain text 429.
```

Comment:
Can you specify "the response from Rack::Attack"?

Reply:
Are we definitely using Rack::Attack for this? I thought it was unrelated since james was doing it on the Systems side. (If it is Rack::Attack, should it be in the codebase with the other Rack::Attack settings?)

Reply:
Yeah we are, it's the default setting.

https://github.com/rack/rack-attack/blob/2257f008763b405f6a1515b460f80ed5d0f06c2c/lib/rack/attack/configuration.rb#L10-L20